### ISSUE TITLE
IINA Crash on macOS Big Sur 11.0 Beta [NSNextStepFrame]

### ISSUE THREAD

**Issue by erickyim** [iina/iina#3177]
**System and IINA version:**
- macOS 11.0 Beta (20A5395g)
- IINA 1.1.0

**Expected behavior:**
IINA Crash.

**Actual behavior:**

<details>
<summary><b>Crash report:</b></summary>

```
21:39:35.737 [player0][d] File loaded
21:39:35.737 [player0][d] Track list changed
2020-10-29 21:39:48.557924+0800 IINA[2028:33289] -[NSNextStepFrame _displayName]: unrecognized selector sent to instance 0x7f94b15098f0
2020-10-29 21:39:48.560546+0800 IINA[2028:33289] [General] An uncaught exception was raised
2020-10-29 21:39:48.560655+0800 IINA[2028:33289] [General] -[NSNextStepFrame _displayName]: unrecognized selector sent to instance 0x7f94b15098f0
2020-10-29 21:39:48.560756+0800 IINA[2028:33289] [General] (
	0   CoreFoundation                      0x00007fff2068a6cf __exceptionPreprocess + 242
	1   libobjc.A.dylib                     0x00007fff203c13c9 objc_exception_throw + 48
	2   CoreFoundation                      0x00007fff2070ccd5 -[NSObject(NSObject) __retain_OA] + 0
	3   CoreFoundation                      0x00007fff205f207d ___forwarding___ + 1467
	4   CoreFoundation                      0x00007fff205f1a38 _CF_forwarding_prep_0 + 120
	5   AppKit                              0x00007fff22f5cbaf -[NSWindow title] + 140
	6   AppKit                              0x00007fff236c4ab6 -[NSWindow _calcAndSetFilenameTitle] + 79
	7   IINA                                0x000000010c569084 $s4IINA20MainWindowControllerC11updateTitleyyF + 2052
	8   IINA                                0x000000010c60d879 $s4IINA22PlayerWindowControllerC13windowDidLoadyyFy10Foundation12NotificationVcfU1_ + 217
	9   IINA                                0x000000010c42f809 $s10Foundation12NotificationVIegn_So14NSNotificationCIeyBy_TR + 153
	10  Foundation                          0x00007fff2136e49b -[__NSObserver _doit:] + 287
	11  CoreFoundation                      0x00007fff20605fec __CFNOTIFICATIONCENTER_IS_CALLING_OUT_TO_AN_OBSERVER__ + 12
	12  CoreFoundation                      0x00007fff206a18cb ___CFXRegistrationPost_block_invoke + 49
	13  Foundation                          0x00007fff2137a74c __NSBLOCKOPERATION_IS_CALLING_OUT_TO_A_BLOCK__ + 7
	14  Foundation                          0x00007fff2137a644 -[NSBlockOperation main] + 98
	15  Foundation                          0x00007fff2137a5d5 __NSOPERATION_IS_INVOKING_MAIN__ + 17
	16  Foundation                          0x00007fff21379886 -[NSOperation start] + 785
	17  Foundation                          0x00007fff2137956a __NSOPERATIONQUEUE_IS_STARTING_AN_OPERATION__ + 17
	18  Foundation                          0x00007fff2137943d __NSOQSchedule_f + 182
	19  libdispatch.dylib                   0x0000000110bcee78 _dispatch_call_block_and_release + 12
	20  libdispatch.dylib                   0x0000000110bd00b0 _dispatch_client_callout + 8
	21  libdispatch.dylib                   0x0000000110be0260 _dispatch_main_queue_callback_4CF + 1107
	22  CoreFoundation                      0x00007fff2064d950 __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 9
	23  CoreFoundation                      0x00007fff2060f852 __CFRunLoopRun + 2731
	24  CoreFoundation                      0x00007fff2060e6ce CFRunLoopRunSpecific + 563
	25  HIToolbox                           0x00007fff2890f0b0 RunCurrentEventLoopInMode + 292
	26  HIToolbox                           0x00007fff2890eeac ReceiveNextEventCommon + 709
	27  HIToolbox                           0x00007fff2890ebcf _BlockUntilNextEventMatchingListInModeWithFilter + 64
	28  AppKit                              0x00007fff22e1fd69 _DPSNextEvent + 883
	29  AppKit                              0x00007fff22e1e52f -[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 1366
	30  AppKit                              0x00007fff22e1095e -[NSApplication run] + 586
	31  AppKit                              0x00007fff22de4d21 NSApplicationMain + 816
	32  IINA                                0x000000010c600fcd main + 13
	33  libdyld.dylib                       0x00007fff20533591 start + 1
)
```
</details>

<br>

<details>
<summary><b>mpv log:</b></summary>

```
```
</details>

<br>

**Steps to reproduce:**
Double click a video file, or Launch IINA and open a video file.

- [ ] MPV does not have this problem.

**How often does this happen?**
Always

**Comment by erickyim** [comment]
Crash Reason: ` Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[NSNextStepFrame _displayName]: unrecognized selector sent to instance 0x7f94b15098f0'`

Seems some Non-public APIs were removed.

See discuss [https://github.com/electron/electron/issues/20027#issue-487040201](https://github.com/electron/electron/issues/20027#issue-487040201)

**Comment by lhc70000** [comment]
I believe that IINA only updates the window title here. It crashed in AppKit itself, which seems to be a system bug.

Does it happen all the time?

**Comment by erickyim** [comment]
> I believe that IINA only updates the window title here. It crashed in AppKit itself, which seems to be a system bug.
> 
> Does it happen all the time?

Yes, it crashes every time.

**Comment by erickyim** [comment]
> I believe that IINA only updates the window title here. It crashed in AppKit itself, which seems to be a system bug.
> 
> Does it happen all the time?

![image](https://user-images.githubusercontent.com/8186665/97585839-88c27900-1a34-11eb-84c0-22ef2f800600.png)

 It not crashes when I uncheck `Enter fullscreen`.

**Comment by saagarjha** [comment]
Based on the crash log, my guess is a UAF probably because we forgot to un-register an observer somewhere. Let me check the code to see what's going on.

**Comment by malhal** [comment]
I get this crash if I double click to enter full screen, then double click to exit full screen, then command+f to enter fullscreen (double click is now broken at this point) then it crashes at the end of the chapter.

**Comment by jeff-h** [comment]
I'm experiencing this crash simply by playing a specific video fullscreen. It doesn't appear to matter how or when I went fullscreen. So to be specific:

* I launch Iina and open the video
* go fullscreen here (either double-click or cmd-opt-f), or maybe while the video is playing. The crash occurs either way.
* skip forward to near the end of the first chapter and press play. Iina crashes at the beginning of the second chapter.

Simplistically it looks like AppKit is crashing when trying to set the window title for a fullscreen window, which doesn't actually display a title.

**Comment by Guillaume-iov42** [comment]
I have the same issue, but only if I use "legacy full screen" (whatever that is).
If you disable it in the preferences (General), the problem goes away. Fine with me.

**Comment by low-batt** [comment]
I too initially was expecting some kind of problem with improper handling of an observer.
That was not it.

See my comment in #3543

**Comment by low-batt** [comment]
The fix in PR #3547 has been merged into the develop branch.

**Comment by erickyim** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Comment by low-batt** [comment]
Closing. Fixed in IINA 1.3.0.

**Comment by erickyim** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。
